popgen/mcmcSelection.go

```diff
@@ -154,3 +180,16 @@ func findMaxN(data AFS) int {
 	}
 	return answer
 }
+
+func cacheIndexToAlleleFrequencyCache(bins int) []float64 {
+	var answer []float64 = make([]float64, bins + 1)
+	//this is done with linear interpolation, as index zero corresponds to an allele frequency of the LeftBound and the index at len[n][k], the number of values p, is the rightBound allele frequency. y = mx + b. First we find m and b.
+	//b is simply the LeftBound.
+	m := (LeftBound - RightBound) / (bins + 1)
+	for i := 0; i < len(bins+1); i++ {
+		answer[i] = float64(i)*m + LeftBound
```

Comment:
I think answer[bins] should equal RightBound, which is why I think "m" should be divided by only "bins"